mh/database.py

```diff
@@ -579,36 +585,40 @@ def find_users(self, capcom_id="", hunter_name=""):
     def create_city(self, session, server_id, gate_id, index,
                     settings, optional_fields):
         city = self.get_city(server_id, gate_id, index)
-        city.optional_fields = optional_fields
+        with city.lock():
+            city.optional_fields = optional_fields
         return city
 
     def join_city(self, session, server_id, gate_id, index):
         city = self.get_city(server_id, gate_id, index)
-        city.parent.players.remove(session)
-        city.players.add(session)
+        with city.lock():
+            city.parent.players.remove(session)
```

Comment:
Why aren't we locking the `parent.players`? Just curious, just want to know your reasoning, in any case I don't think it's a big deal.

Reply:
`Players.remove()` and `Players.add()` already have locks inside of them, but you're right. Technically they should be unified here to ensure the lock is not dropped between player removal and player addition. That is out of scope of this PR though, being solely related to the `Players` class, so if it is to be fixed it should happen in another separate PR.